h3/src/quic.rs

```diff
@@ -121,7 +122,7 @@ pub trait SendStream<B: Buf> {
     fn reset(&mut self, reset_code: u64);
 
     /// Get QUIC send stream id
-    fn id(&self) -> u64;
+    fn id(&self) -> StreamId;
```

Comment:
This is the public trait that people implement to integrate a QUIC library... Should we require the stronger type here? It seems we could continue to just allow it to be `u64`, and convert internally. Unless there's a strong reason for people to use it the type themselves, being more conservative may be advantageous.

Reply:
I made `StreamId` constructible only via `TryFrom` in a new commit. But I'm not sure if we shouldn't rather `panic!` when an implementer tries to build an invalid StreamId.

Reply:
Yea, I don't think the trait should have `fn id(&self) -> Result<..>`. The ID of the stream should available easily and immediately. What does it mean for a stream to not be able to tell `h3` its valid ID?

Reply:
In  `quic::SendStream::stream_id()`, it means that the quic implementation isn't compliant. So I made it panic there.